RIOT-OS/RIOT
# netopt: redefine NETOPT_LINK_CONNECTED

### Contribution description
This PR redefines the behavior of the `NETOPT_LINK_CONNECTED` (#8601) command to be able to get the link status of any network interface and set the link up or down.

In a nutshell:
- It renames the command to NETOPT_LINK
- It changes the documentation to reflect that it can be used to set the link state.
- It changes the option description from "PHY link" to "network interface link", so it can be used to represent any kind of link (Ethernet, LoRaWAN, NB-IoT, etc).
- Adds the corresponding `ifconfig` command for setting link up and down

This code is refactored from #11022. As discussed with @miri64, it's better to add it generically so other connection oriented interfaces can reuse it.



### Testing procedure
It can be tested with `examples/default`.
- Check `ifconfig help` documentation (it should show `ifconfig <if_id> [up|down]`
- Try to set an interface up or down. Although nobody uses this NETOPT_LINK command yet, it should be possible to check that the error messages (-ENOTSUP) should match `up` or `down`.



### Issues/PRs references
None so far
